x-pack/plugins/spaces/public/share_saved_objects_to_space/components/share_to_space_flyout_internal.test.tsx

```diff
@@ -0,0 +1,745 @@
+      savedObjectToShare,
+    } = await setup();
+
+    expect(wrapper.find(ShareToSpaceForm)).toHaveLength(1);
+    expect(wrapper.find(EuiLoadingSpinner)).toHaveLength(0);
+    expect(wrapper.find(NoSpacesAvailable)).toHaveLength(0);
+
+    // Using props callback instead of simulating clicks,
+    // because EuiSelectable uses a virtualized list, which isn't easily testable via test subjects
+    const spaceSelector = wrapper.find(SelectableSpacesControl);
+
+    act(() => {
+      spaceSelector.props().onChange([]);
+    });
+
+    const startButton = findTestSubject(wrapper, 'sts-initiate-button');
+
+    await act(async () => {
+      startButton.simulate('click');
+      await nextTick();
+      wrapper.update();
+    });
+
+    const { type, id } = savedObjectToShare;
+    const { shareSavedObjectAdd, shareSavedObjectRemove } = mockSpacesManager;
+    expect(shareSavedObjectAdd).not.toHaveBeenCalled();
+    expect(shareSavedObjectRemove).toHaveBeenCalledWith({ type, id }, ['space-1']);
+
+    expect(mockToastNotifications.addSuccess).toHaveBeenCalledTimes(1);
+    expect(mockToastNotifications.addError).not.toHaveBeenCalled();
+    expect(onClose).toHaveBeenCalledTimes(1);
+  });
+
+  it('allows the form to be filled out to add and remove a space', async () => {
+    const {
+      wrapper,
+      onClose,
+      mockSpacesManager,
+      mockToastNotifications,
+      savedObjectToShare,
+    } = await setup();
+
+    expect(wrapper.find(ShareToSpaceForm)).toHaveLength(1);
+    expect(wrapper.find(EuiLoadingSpinner)).toHaveLength(0);
+    expect(wrapper.find(NoSpacesAvailable)).toHaveLength(0);
+
+    // Using props callback instead of simulating clicks,
+    // because EuiSelectable uses a virtualized list, which isn't easily testable via test subjects
+    const spaceSelector = wrapper.find(SelectableSpacesControl);
+
+    act(() => {
+      spaceSelector.props().onChange(['space-2', 'space-3']);
+    });
+
+    const startButton = findTestSubject(wrapper, 'sts-initiate-button');
+
+    await act(async () => {
+      startButton.simulate('click');
+      await nextTick();
+      wrapper.update();
+    });
+
+    const { type, id } = savedObjectToShare;
+    const { shareSavedObjectAdd, shareSavedObjectRemove } = mockSpacesManager;
+    expect(shareSavedObjectAdd).toHaveBeenCalledWith({ type, id }, ['space-2', 'space-3']);
+    expect(shareSavedObjectRemove).toHaveBeenCalledWith({ type, id }, ['space-1']);
+
+    expect(mockToastNotifications.addSuccess).toHaveBeenCalledTimes(2);
+    expect(mockToastNotifications.addError).not.toHaveBeenCalled();
+    expect(onClose).toHaveBeenCalledTimes(1);
+  });
+
+  describe('correctly renders checkable cards', () => {
+    function getCheckableCardProps(
+      wrapper: ReactWrapper<React.PropsWithChildren<EuiCheckableCardProps>>
+    ) {
+      const iconTip = wrapper.find(EuiIconTip);
+      return {
+        checked: !!wrapper.prop('checked'),
+        disabled: !!wrapper.prop('disabled'),
```

Comment:
__nit:__ looks like these are `boolean` already (per TS):
```suggestion
        checked: wrapper.prop('checked'),
        disabled: wrapper.prop('disabled'),
```